src/jit/morph.cpp

```diff
@@ -15564,42 +15564,46 @@ void                Compiler::fgPromoteStructs()
     //
 
     lvaStructPromotionInfo structPromotionInfo;
+    bool tooManyLocals = false;
 
     for (unsigned lclNum = 0;
-        lclNum < startLvaCount;
-        lclNum++)
+         lclNum < startLvaCount;
+         lclNum++)
     {
+        bool canPromote = false;
         LclVarDsc*  varDsc = &lvaTable[lclNum];
 
+#ifdef FEATURE_SIMD
+        if (varDsc->lvSIMDType && varDsc->lvUsedInSIMDIntrinsic)
+        {
+            // If we have marked this as lvUsedInSIMDIntrinsic, then we do not want to promote
+            // its fields.  Instead, we will attempt to enregister the entire struct.
+            varDsc->lvRegStruct = true;
+        }
+        else
+#endif //FEATURE_SIMD
         // Don't promote if we have reached the tracking limit.
         if (lvaHaveManyLocals())
         {
-            JITDUMP("Stopped promoting struct fields, due to too many locals.\n");
-            break;
+            // Print the message first time when we detected this condition
+            if (!tooManyLocals)
+            {
+                JITDUMP("Stopped promoting struct fields, due to too many locals.\n");
+            }
+            tooManyLocals = true;
         }
 #if !FEATURE_MULTIREG_STRUCT_PROMOTE
-        if (varDsc->lvIsMultiRegArgOrRet)
+        else if (varDsc->lvIsMultiRegArgOrRet)
         {
             JITDUMP("Skipping V%02u: marked lvIsMultiRegArgOrRet.\n", lclNum);
-            continue;
         }
 #endif // !FEATURE_MULTIREG_STRUCT_PROMOTE
-
-#ifdef FEATURE_SIMD
-        if (varDsc->lvSIMDType && varDsc->lvUsedInSIMDIntrinsic)
-        {
-            // If we have marked this as lvUsedInSIMDIntrinsic, then we do not want to promote
-            // its fields.  Instead, we will attempt to enregister the entire struct.
-            // Note, however, that if the code below does not decide to promote this struct,
-            // we will still set lvRegStruct if its fields have not been accessed.
-            varDsc->lvRegStruct = true;
-        }
-        else
-#endif // FEATURE_SIMD
-        if (varTypeIsStruct(varDsc))
+        else if (varTypeIsStruct(varDsc))
         {
             lvaCanPromoteStructVar(lclNum, &structPromotionInfo);
-            if (structPromotionInfo.canPromote)
+            canPromote = structPromotionInfo.canPromote;
+
+            if (canPromote)
             {
                 // We *can* promote; *should* we promote?
```

Comment:
Based upon these comments, I think that it would be best to introduce a new bool predicate here:
bool  shouldlPromote = true;   //  This get set to false if we decide that it is not useful to promote this variable